examples/elasto_plastic.jl

```diff
@@ -0,0 +1,300 @@
+using JuAFEM
+using Tensors
+using KrylovMethods
+using TimerOutputs
+using Statistics: mean
+import ProgressMeter
+const ∇ = Tensors.gradient
+
+struct Material{T}
+    μ::Float64		#Lamé constant
+    λ::Float64		#Lamé constant
+    τ₀::Float64		#Yield stress
+    H::Float64		#Hardening
+end
+
+struct MaterialState{T}
+	S::SymmetricTensor{2,3,T,6} #2nd PK stress
+    ϵᵖ::Float64					#Plastic strain
+    F⁻¹::Tensor{2,3,T,9}		#Plastic deformation grad
+    ν::Tensor{2,3,T,9}			
+end
+
+function MaterialState()
+    return MaterialState(
+    				zero(SymmetricTensor{2,3,Float64,6}),
+    				0.0, 
+    	            one(Tensor{2,3,Float64,9}), 
+    	            zero(Tensor{2,3,Float64,9}),)
+end
+
+function _compute_2nd_PK(mp::Material, E, state::MaterialState)
+    μ, λ, τ₀, H = mp.μ, mp.λ, mp.τ₀, mp.H
+    
+    ϵᵖ = state.ϵᵖ
+    F⁻¹ = state.F⁻¹
+    ν = state.ν
+    dγ = 0 
+
+    C = 2E + one(E)
+
+    phi, Mdev, Stilde = ϕ(C, F⁻¹, ϵᵖ, μ, λ, τ₀, H)
+
+    if phi > 0 #Plastic step
+        dγ = 0.0
+        
+        #Setup newton varibles
+        TOL = 0.00001
+        newton_error, counter = TOL + 1, 0
+
+        # Define residual for semi-emplicit backward euler (assuming ⁿ⁺¹ν = ⁿν)
+        ϕ_residual(dγ) = ϕ(C, F⁻¹ - dγ*F⁻¹⋅ν, ϵᵖ + dγ, μ, λ, τ₀, H)[1]
+
+        while newton_error > TOL; counter +=1 
+            
+            res = ϕ_residual(dγ)
+
+            #Not able to take Automatic derivative of ϕ_residual for some reaseon
+            #Use numerical derivative for simplicity
+
+            #jac = ForwardDiff.derivative(ϕ_residual, dγ)
+            jac = (ϕ_residual(dγ + 0.00001) - ϕ_residual(dγ))/(0.00001)
+
+            ddγ = jac\-res
+            dγ += ddγ
+
+            newton_error = norm(res)
+
+            if counter > 10
+                error("Could not find equilibrium in material")
+            end
+        end
+
+        ϵᵖ += dγ
+        F⁻¹ -= dγ*F⁻¹⋅ν
+
+        _, Mdev, Stilde = ϕ(C, F⁻¹, ϵᵖ, μ, λ, τ₀, H)
+    end
+
+    dϕdM = Mdev/ (1e-15 + sum(sum(Mdev⋅Mdev)))
+    ν = sqrt(3/2)*dϕdM
+    S = symmetric(F⁻¹⋅Stilde⋅F⁻¹')
+
+    return S, ϵᵖ, ν, F⁻¹
+end
+
+#Yield function
+function ϕ(C, F⁻¹, ϵᵖ, μ, λ, τ₀, H)
+    Cᵉ = F⁻¹'⋅C⋅F⁻¹
+
+    Stilde =  μ*(one(Cᵉ) - inv(Cᵉ)) + λ*log(sqrt(det(Cᵉ))) * inv(Cᵉ)	#Neo Hook
+    #Stilde = λ/2*(tr(Cᵉ)-3)*one(Cᵉ) + μ*(Cᵉ - one(Cᵉ)) 				#St Venant-Kirchhoff
+
+    #Mandel stress
+    Mbar = Cᵉ⋅Stilde
+    Mdev = dev(Mbar)
+
+    yield_func = sqrt(3/2)*norm(Mdev) - (τ₀ + H*ϵᵖ)
+    
+    return yield_func, Mdev, Stilde
+end
+
+function constitutive_driver(mp::Material, E, state::MaterialState)
+
+	#Is it possible to combine the computation for S and ∂S∂E (using auto diff?)
+    S, ϵᵖ, ν, F⁻¹ = _compute_2nd_PK(mp, E, state)
+    ∂S∂E, _ = ∇(E -> _compute_2nd_PK(mp, E, state)[1], E, :all)
```

Comment:
I am not able to get `∂S∂E` and `S` at the same time, since `_compute_2nd_PK` exports multiple values